PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/functions/run/AzureFunctionsHostConfigurationParameters.kt

```diff
@@ -0,0 +1,225 @@
+        private const val PROJECT_TFM = "PROJECT_TFM"
+        const val PROJECT_NOT_SPECIFIED = "Project is not specified"
+        const val SOLUTION_IS_LOADING = "Solution is loading, please wait for a few seconds"
+
+        private val logger = getLogger<AzureFunctionsHostConfigurationParameters>()
+    }
+
+    val isUnloadedProject: Boolean
+        get() = project.getComponent<ProjectModelViewHost>().isUnloadedProject(projectFilePath)
+
+    override fun toDotNetExecutable(): DotNetExecutable {
+        val runnableProject = tryGetRunnableProject() ?: throw CantRunException("Project is not specified")
+        val projectOutput = tryGetProjectOutput(runnableProject)
+
+        val coreToolsInfo: FunctionsCoreToolsInfo? = FunctionsCoreToolsInfoProvider.build()
+                ?: throw CantRunException("Can't run Azure Functions host - path to core tools has not been configured.")
+
+        val effectiveWorkingDirectory = if (trackProjectWorkingDirectory && projectOutput != null) {
+            projectOutput.workingDirectory
+        } else {
+            workingDirectory
+        }
+        val effectiveArguments = if (trackProjectArguments && projectOutput != null && projectOutput.defaultArguments.isNotEmpty()) {
+            projectOutput.defaultArguments
+        } else {
+            ParametersListUtil.parse(programParameters)
+        }
+
+        return DotNetExecutable(
+                exePath = coreToolsInfo!!.coreToolsExecutable,
+                projectTfm = projectOutput?.tfm ?: projectTfm,
+                workingDirectory = effectiveWorkingDirectory,
+                programParameters = effectiveArguments,
+                useMonoRuntime = useMonoRuntime,
+                useExternalConsole = useExternalConsole,
+                environmentVariables = envs,
+                isPassParentEnvs = isPassParentEnvs,
+                onProcessStarter = startBrowserAction,
+                assemblyToDebug = coreToolsInfo.coreToolsExecutable,
+                runtimeArguments = runtimeArguments,
+                executeAsIs = true)
+    }
+
+    private val startBrowserAction: (RunProfile, ProcessHandler) -> Unit = { runProfile, processHandler ->
+        if (startBrowserParameters.startAfterLaunch && runProfile is RunConfiguration) {
+            val startBrowserSettings = StartBrowserSettings().apply {
+                isSelected = startBrowserParameters.startAfterLaunch
+                url = startBrowserParameters.url
+                browser = startBrowserParameters.browser
+                isStartJavaScriptDebugger = startBrowserParameters.withJavaScriptDebugger
+            }
+
+            BrowserStarter(runProfile, startBrowserSettings, processHandler).start()
+        }
+    }
+
+    private fun tryGetRunnableProject(): RunnableProject? {
+        val runnableProjects = project.solution.runnableProjectsModel.projects.valueOrNull
+        if (runnableProjects != null) {
```

Comment:
You can use Elvis operator for such constructions: 
`... .valueOrNull ?: return null`